### PR TITLE
Creating 1.2.x branch, changing related links

### PR DESCRIPTION
**Pull request by GregorioDiStefano**
Signed-off-by: Greg Di Stefano <greg.distefano+github@gmail.com>

### CODE REVIEW

**Comment by GregorioDiStefano** [comment]
Here are the remaining references to master, that I left untouched:

```

202.Release-information/02.Open-source-licenses/docs.md-
202.Release-information/02.Open-source-licenses/docs.md-github.com/pelletier/go-buffruneio:
202.Release-information/02.Open-source-licenses/docs.md-
202.Release-information/02.Open-source-licenses/docs.md-# buffruneio
202.Release-information/02.Open-source-licenses/docs.md-
202.Release-information/02.Open-source-licenses/docs.md:[![Tests Status](https://travis-ci.org/pelletier/go-buffruneio.svg?branch=master)](https://travis-ci.org/pelletier/go-buffruneio)
202.Release-information/02.Open-source-licenses/docs.md-[![GoDoc](https://godoc.org/github.com/pelletier/go-buffruneio?status.svg)](https://godoc.org/github.com/pelletier/go-buffruneio)
202.Release-information/02.Open-source-licenses/docs.md-
202.Release-information/02.Open-source-licenses/docs.md-Buffruneio is a wrapper around bufio to provide buffered runes access with
202.Release-information/02.Open-source-licenses/docs.md-unlimited unreads.
202.Release-information/02.Open-source-licenses/docs.md-
--
06.Administration/06.Upgrading/docs.md-> remote: Total 367 (delta 134), reused 122 (delta 122), pack-reused 214  
06.Administration/06.Upgrading/docs.md-> Receiving objects: 100% (367/367), 83.55 KiB | 0 bytes/s, done.  
06.Administration/06.Upgrading/docs.md-> Resolving deltas: 100% (214/214), completed with 42 local objects.  
06.Administration/06.Upgrading/docs.md-> From https://github.com/mendersoftware/integration  
06.Administration/06.Upgrading/docs.md->    02cd118..75b7831  1.0.x      -> origin/1.0.x  
06.Administration/06.Upgrading/docs.md:>    06f3212..e9e5df4  master     -> origin/master  
06.Administration/06.Upgrading/docs.md-
06.Administration/06.Upgrading/docs.md-For each release there will be a corresponding release branch. For example, the
06.Administration/06.Upgrading/docs.md-branch named `1.0.x` provides the 1.0 release setup. Stable releases are tagged,
06.Administration/06.Upgrading/docs.md-e.g. `1.0.1`.
06.Administration/06.Upgrading/docs.md-
--
03.Devices/03.Integrating-with-U-Boot/docs.md-
03.Devices/03.Integrating-with-U-Boot/docs.md-## Practical example
03.Devices/03.Integrating-with-U-Boot/docs.md-
03.Devices/03.Integrating-with-U-Boot/docs.md-For a real life example of a patch used to integrate with Mender, check out [the
03.Devices/03.Integrating-with-U-Boot/docs.md-patch for
03.Devices/03.Integrating-with-U-Boot/docs.md:BeagleBone](https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-beaglebone/recipes-bsp/u-boot/patches/0001-BBB-Use-Mender-boot-code-for-selecting-boot-device-a.patch?target=_blank)
03.Devices/03.Integrating-with-U-Boot/docs.md-on github.com. This patch displays all of the steps required to patch the U-Boot
03.Devices/03.Integrating-with-U-Boot/docs.md-boot code for BeagleBone, including most of the steps described on this page.
--
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md-@@ -9,3 +12,30 @@
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md- SRCREV = "df61a74e6845ec9bdcdd48d2aff5e9c2c6debeaa"
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md- 
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md- PV = "v2016.03+git${SRCPV}"
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md-+
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md:+SRC_URI = "git://git.denx.de/u-boot.git;branch=master"
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md-+
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md-+S = "${WORKDIR}/git"
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md-+
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md-+INSANE_SKIP_${PN} = "already-stripped"
03.Devices/03.Integrating-with-U-Boot/01.Providing-custom-u-boot-fw-utils/docs.md-+EXTRA_OEMAKE_class-target = 'CROSS_COMPILE=${TARGET_PREFIX} CC="${CC} ${CFLAGS} ${LDFLAGS}" V=1'
--
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md-2. Second, any existing `CONFIG_BOOTCOUNT_LIMIT` and/or `CONFIG_BOOTCOUNT_ENV`
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md-   defines in the configuration must be removed.
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md-
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md-For an example, take a look at [the patch provided for the `vexpress-qemu`
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md-`MACHINE`
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md:type](https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-qemu/recipes-bsp/u-boot/patches/experimental/0002-Enable-custom-bootlimit-code-for-vexpress-qemu.patch).
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md-You can try this patch by adding this to your `u-boot-my-fork.bb` or
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md-`u-boot-my-fork.bbappend` file:
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md-
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md-```bash
03.Devices/03.Integrating-with-U-Boot/02.U-Boot-versions-without-BOOTLIMIT-support/docs.md-FILESEXTRAPATHS_prepend := "${THISDIR}/../../../meta-mender-qemu/recipes-bsp/u-boot/patches/experimental:"
--
03.Devices/05.Raw-flash/docs.md-### Mender
03.Devices/05.Raw-flash/docs.md-
03.Devices/05.Raw-flash/docs.md-The Mender Yocto layer comes with support for the following raw flash boards in
03.Devices/05.Raw-flash/docs.md-the main tree:
03.Devices/05.Raw-flash/docs.md-
03.Devices/05.Raw-flash/docs.md:* [Toradex Colibri VF50/VF61 `meta-mender-toradex-nxp`](https://github.com/mendersoftware/meta-mender/tree/master/meta-mender-toradex-nxp)
03.Devices/05.Raw-flash/docs.md:* [vexpress-a9 emulated by QEMU `meta-mender-qemu`](https://github.com/mendersoftware/meta-mender/tree/master/meta-mender-qemu)
03.Devices/05.Raw-flash/docs.md-
03.Devices/05.Raw-flash/docs.md-Both layers provide a set of patches based on their respective upstream U-Boot
03.Devices/05.Raw-flash/docs.md-sources and can be used as a reference when implementing your raw flash based
03.Devices/05.Raw-flash/docs.md-machine.
03.Devices/05.Raw-flash/docs.md-
--
03.Devices/05.Raw-flash/docs.md-
03.Devices/05.Raw-flash/docs.md-By default a `*.ubifs` root filesystem image will be used when generating a Mender artifact.
03.Devices/05.Raw-flash/docs.md-
03.Devices/05.Raw-flash/docs.md-### meta-mender integration
03.Devices/05.Raw-flash/docs.md-
03.Devices/05.Raw-flash/docs.md:By inheriting [`mender-install-ubi`](https://github.com/mendersoftware/meta-mender/blob/master/meta-mender-core/classes/mender-install-ubi.bbclass) (included
03.Devices/05.Raw-flash/docs.md-in `mender-full-ubi`) the following configuration settings will be set
03.Devices/05.Raw-flash/docs.md-automatically:
03.Devices/05.Raw-flash/docs.md-
03.Devices/05.Raw-flash/docs.md-* `MENDER_STORAGE_DEVICE` - defaults to `ubi0`
03.Devices/05.Raw-flash/docs.md-* `MENDER_ROOTFS_PART_A` - defaults to `ubi0_0`
--
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-!!! For general information about getting started with Yocto Project, it is recommended to read the [Yocto Project Quick Start guide](http://www.yoctoproject.org/docs/2.2/yocto-project-qs/yocto-project-qs.html?target=_blank).
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-## Prerequisites
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md:! We use the Yocto Project's **master** branch below. *Building meta-mender on other releases of the Yocto Project will likely not work seamlessly.* We use the `master` branch in `meta-mender`, which builds a latest release of Mender for the bleeding edge Yocto Project revision. `meta-mender` also has other branches like [daisy](https://github.com/mendersoftware/meta-mender/tree/daisy?target=_blank) that correspond to Yocto Project releases , but these branches are no longer maintained by Mender developers. Please reach out on the [Mender community mailing list](https://groups.google.com/a/lists.mender.io/forum?target=_blank#!forum/mender) if you would like help with getting Mender to work on other versions of the Yocto Project.
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-!!! The meta-mender-demo layer, which is used below, and the web-server, are bundled with a default demo certificate and key. If you are intending on using Mender in production, you must generate your own certificate using OpenSSL. Please see the certificate section [for the server](../../administration/certificates-and-keys) and [for the client](../building-for-production/#certificates) for more information.
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-The required meta layers are found in the following repositories:
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-```
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-URI: git://git.yoctoproject.org/poky
04.Artifacts/01.Building-Mender-Yocto-image/docs.md:branch: master
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-URI: git://github.com/mendersoftware/meta-mender
04.Artifacts/01.Building-Mender-Yocto-image/docs.md:branch: master
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-```
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-A Yocto Project poky environment is required. If you already have 
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-this in your build environment, please open a terminal, go to the `poky`
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-directory and skip to [Adding the meta layers](#adding-the-meta-layers).
--
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-On the other hand, if you want to start from a *clean Yocto Project environment*,
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-you need to clone the latest poky and go into the directory:
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-```bash
04.Artifacts/01.Building-Mender-Yocto-image/docs.md:git clone -b master git://git.yoctoproject.org/poky
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-```
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-```bash
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-cd poky
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-```
--
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-We will now add the required meta layers to our build environment.
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-Please make sure you are standing in the directory where `poky` resides,
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-i.e. the top level of the Yocto Project build tree, and run these commands:
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-```bash
04.Artifacts/01.Building-Mender-Yocto-image/docs.md:git clone -b master git://github.com/mendersoftware/meta-mender
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-```
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-Next, we initialize the build environment:
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-
04.Artifacts/01.Building-Mender-Yocto-image/docs.md-```bash
--
chapter.md-
chapter.md-### Development
chapter.md-
chapter.md-# Documentation
chapter.md-
chapter.md:Documentation for Mender's bleeding-edge master branch 
chapter.md-
chapter.md:[https://github.com/mendersoftware/mender/tree/master](https://github.com/mendersoftware/mender/tree/master)
```

File: 01.Getting-started/03.Deploy-to-virtual-devices/docs.md [hunk 1]
```diff
@@ -68,7 +68,7 @@ can be used, and steps to build one are provided at
 
 To make testing easier, a Mender Artifact that can be used with
 the virtual device is provided for download at
-[https://d1b0l86ne08fsf.cloudfront.net/master/vexpress-qemu/vexpress_release_2.mender](https://d1b0l86ne08fsf.cloudfront.net/master/vexpress-qemu/vexpress_release_2.mender).
+[https://s3.amazonaws.com/mender/1.2.0b1-build3/vexpress-qemu/vexpress_release_2_1.2.0b1-build3.mender](https://s3.amazonaws.com/mender/1.2.0b1-build3/vexpress-qemu/vexpress_release_2_1.2.0b1-build3.mender).
```
Comment (estenberg) [inline 01.Getting-started/03.Deploy-to-virtual-devices/docs.md]: Please use cloudfront, not s3 directly. This seems to happen on every release.. is this from some script that needs updating?

File: 01.Getting-started/03.Deploy-to-virtual-devices/docs.md [hunk 2]
```diff
@@ -119,7 +119,7 @@ uploaded, Mender will see that it contains the same rootfs
 that is already installed and skip the deployment. It will
 immediately be marked as successful and moved to *Past deployments*.
 For this reason, we provide another Artifact that you can use
-to deploy with at [https://d1b0l86ne08fsf.cloudfront.net/master/vexpress-qemu/vexpress_release_1.mender](https://d1b0l86ne08fsf.cloudfront.net/master/vexpress-qemu/vexpress_release_1.mender).
+to deploy with at [https://s3.amazonaws.com/mender/1.2.0b1-build3/vexpress-qemu/vexpress_release_1_1.2.0b1-build3.mender](https://s3.amazonaws.com/mender/1.2.0b1-build3/vexpress-qemu/vexpress_release_1_1.2.0b1-build3.mender).
```
Comment (estenberg) [inline 01.Getting-started/03.Deploy-to-virtual-devices/docs.md]: Cloudfront, please.

File: 01.Getting-started/03.Deploy-to-virtual-devices/docs.md (hunk 1 above)
Comment (GregorioDiStefano) [inline 01.Getting-started/03.Deploy-to-virtual-devices/docs.md]: Thanks, failed copy and paste.

**Comment by GregorioDiStefano** [comment]
All fixed, thanks!

**Comment by drewmoseley** [comment]
@estenberg is it possible to use a dead link checker before this deployed to the live server?

**Comment by GregorioDiStefano** [comment]
@drewmoseley @estenberg I actually checked for dead links using the following:

```
files=`find . -iname "*.md"`
echo $files | xargs -L 1 markdown-link-check -p
```

You can find the tool here: https://www.npmjs.com/package/markdown-link-check

**Comment by drewmoseley** [comment]
OK.  I played with that a bit yesterday but definitely thanks for the full command line.

**Comment by drewmoseley** [comment]
Looks good for me now.  @estenberg ok to merge?

**Comment by estenberg** [comment]
Yep! Merging! Thank you!

**Pull request merged**